ui/app/AppLayouts/Profile/views/AdvancedView.qml

```diff
@@ -386,30 +386,55 @@ ScrollView {
                 }
             }
 
-            // TODO: replace with StatusQ component
+            StatusSectionHeadline {
+                text: qsTr("Developer features")
+                topPadding: Style.current.bigPadding
+                bottomPadding: Style.current.padding
+            }
+
+            Separator {
+                anchors.topMargin: Style.current.bigPadding
+                anchors.left: parent.left
+                anchors.leftMargin: -Style.current.padding
+                anchors.right: parent.right
+                anchors.rightMargin: -Style.current.padding
+            }
+
             StatusSettingsLineButton {
-                text: qsTr("Download messages")
+                text: qsTr("Full developer mode")
                 isSwitch: true
-                switchChecked: localAccountSensitiveSettings.downloadChannelMessagesEnabled
+                visible: {
+                    return !localAccountSensitiveSettings.downloadChannelMessagesEnabled ||
+                        !root.store.profileModuleInst.isTelemetryEnabled ||
+                        !root.store.profileModuleInst.isDebugEnabled ||
+                        !root.store.profileModuleInst.isAutoMessageEnabled
+                }
                 onClicked: {
-                    localAccountSensitiveSettings.downloadChannelMessagesEnabled = !localAccountSensitiveSettings.downloadChannelMessagesEnabled
+                    openPopup(enableDeveloperFeaturesConfirmationDialogComponent, {light: false})
```

Comment:
There doesn't seem to be a `light` property on that popup. Sure this should be here?

Reply:
hummm, I am guilty of blindly copy/pasting this code from somewhere else, let me confirm that and remove if the property doesn't exists

Reply:
Removed, indeed good catch